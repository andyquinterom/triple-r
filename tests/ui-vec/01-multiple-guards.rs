use triple_r::ReusableVec;

fn main() {
    let mut vec = ReusableVec::<i32>::default();
    let _guard1 = vec.recycle::<i32>();
    let _guard2 = vec.recycle::<i32>();
}
