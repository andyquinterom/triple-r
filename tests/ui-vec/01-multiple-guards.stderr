error[E0499]: cannot borrow `vec` as mutable more than once at a time
 --> tests/ui-vec/01-multiple-guards.rs:6:19
  |
5 |     let _guard1 = vec.recycle::<i32>();
  |                   --- first mutable borrow occurs here
6 |     let _guard2 = vec.recycle::<i32>();
  |                   ^^^ second mutable borrow occurs here
7 | }
  | - first borrow might be used here, when `_guard1` is dropped and runs the `Drop` code for type `ReusableVecGuard`
